jax/interpreters/pxla.py

```diff
@@ -1172,12 +1225,31 @@ def __init__(self, handlers, out_specs, out_indices, unmapped_local_out_avals):
   def __call__(self, out_bufs):
     return [h(bufs) for h, bufs in safe_zip(self.handlers, out_bufs)]
 
-def avals_to_results_handler(nrep, npart, out_specs, unmapped_local_out_avals):
+
+def avals_to_results_handler(
+    nrep,
+    npart,
+    out_specs,
+    unmapped_local_out_avals,
+    global_out_avals: Optional[Sequence[ShapedArray]] = None,
+    out_axis_resources: Optional[Sequence[AxisResource]] = None,
+    global_mesh=None):
   out_indices = [spec_to_indices(aval.shape, spec)
                  if aval is not core.abstract_unit else None
                  for aval, spec in safe_zip(unmapped_local_out_avals, out_specs)]  # pytype: disable=attribute-error
-  handlers = [aval_to_result_handler(spec, idcs, aval)
-              for spec, idcs, aval in safe_zip(out_specs, out_indices, unmapped_local_out_avals)]
+  if global_out_avals and out_axis_resources and global_mesh:
```

Comment:
I'd suggesting adding `is not None` here, because the current test also fails for empty collections and that seems a little counterintuitive to me.